docs/add-a-new-resource.md

```diff
@@ -0,0 +1,71 @@
+# Adding a New Resource
+
+New resources are required when AWS adds a new service, or adds new features within an existing service which would require a new resource to manage in Terraform. Typically anything with a new set of CRUD API endpoints is a great candidate for a new resource.
+
+Each resource should be submitted for review in isolation, pull requests containing multiple resources are harder to review and the maintainers will normally ask for them to be broken apart.
+
+Please use the `skaff` tool to generate new resource and test templates for any new resource. Doing so will ensure that any boilerplate code, structural best practices and repetitive naming is done for you and always represents our most current standards.
+
+## Prerequisites
+
+If this is the first resource for a new service, please ensure the Service Client for the new service has been added and merged. See [Adding a new Service](add-a-new-service.md) for details.
+
+## Steps to Add a Resource
+
+### Fork the provider and create a feature branch
+
+For a new resources use a branch named `f-{resource name}` for example: `f-ec2-vpc`. See [Raising a Pull Request](raising-a-pull-request.md) for more details.
+
+### Name the resource
+
+Either by creating the file manually, or using `skaff` to generate a template.
+
+All resources should be named with the following pattern: `aws_<service>_<name>`
+
+Where `<service>` is the AWS short service name that matches the key in the `serviceData` map in the `conns` package (created via the [Adding a new Service](add-a-new-service.md))
+
+Where `<name>` represents the conceptual infrastructure represented by the create, read, update, and delete methods of the service API. It should be a singular noun. For example, in an API that has methods such as `CreateThing`, `DeleteThing`, `DescribeThing`, and `ModifyThing` the name of the resource would end in `_thing`.
+
+### Fill out the Resource Schema
+
+In the `internal/service/<service>/<service>.go` file you will see a `Schema` property which exists as a map of `Schema` objects. This relates the AWS API data model with the Terraform resource itself. For each property you want to make available in Terraform, you will need to add it as an attribute, choose the correct data type and supply the correct [Schema Behaviors](https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors) in order to ensure Terraform knows how to correctly handle the value.
+
+Typically you will add arguments to represent the values that are under control by Terraform, and attributes in order to supply read-only values as references for Terraform. These are distinguished by Schema Behavior.
+
+Attribute names are to specified in `camel_case` as opposed to the AWS API which is `CamelCase`
```

Comment:
This duplicates explanation in `skaff`. I always like when there's just one place for something to ease maintenance when it changes. Maybe here we give a couple links and refer them to the `skaff` template?

Reply:
I'll probably lean to just keep this part very high-level, but linking right to the template for details.